### PR TITLE
Chore: Convert tsdb request test to standard library

### PR DESCRIPTION
**Pull request by ceh**
**What this PR does / why we need it**:

Convert tsdb request test to standard library.

**Special notes for your reviewer**:
Part of #22844.

Renamed the test to `TestHandleRequest` because that's the method under test. Also renamed the file to `request_test.go` because `HandleRequest` is defined in https://github.com/grafana/grafana/blob/c4e141c112f797a9c4724311e9617157964f5c18/pkg/tsdb/request.go#L12, which ought to make it a bit easier to find any related test code.

What do you think?

### CODE REVIEW

File: pkg/tsdb/request_test.go [hunk 1]
```diff
@@ -41,23 +38,22 @@ func TestMetricQuery(t *testing.T) {
...
+	t.Run("Should return error when handling request for query with unknown type", func(t *testing.T) {
 		req := &TsdbQuery{
 			Queries: []*Query{
 				{RefId: "A", DataSource: &models.DataSource{Id: 1, Type: "asdasdas"}},
 			},
 		}
 
 		_, err := HandleRequest(context.TODO(), &models.DataSource{Id: 12, Type: "testjughjgjg"}, req)
-		So(err, ShouldNotBeNil)
+		require.NotNil(t, err)
```
Comment (aknuds1) [inline pkg/tsdb/request_test.go:56]:
```suggestion
		require.Err(t, err)
```

**Review by aknuds1 — APPROVED** [review]
Looks great, except for an error assertion, where I think it's more precise to use `require.Error`.

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Looks like I can't apply my suggestion myself before merging, so please just commit it before I re-review :)

File: pkg/tsdb/request_test.go [hunk 2]
```diff
@@ -20,15 +20,12 @@ func TestMetricQuery(t *testing.T) {
 		fakeExecutor.Return("A", TimeSeriesSlice{&TimeSeries{Name: "argh"}})
 
 		res, err := HandleRequest(context.TODO(), &models.DataSource{Id: 1, Type: "test"}, req)
-		So(err, ShouldBeNil)
-
-		Convey("Should return query results", func() {
-			So(res.Results["A"].Series, ShouldNotBeEmpty)
-			So(res.Results["A"].Series[0].Name, ShouldEqual, "argh")
-		})
+		require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/tsdb/request_test.go:23]:
```suggestion
		require.NoError(t, err)
```

File: pkg/tsdb/request_test.go [hunk 3]
```diff
@@ -41,23 +38,22 @@ func TestMetricQuery(t *testing.T) {
 		fakeExecutor.Return("B", TimeSeriesSlice{&TimeSeries{Name: "barg"}})
 
 		res, err := HandleRequest(context.TODO(), &models.DataSource{Id: 1, Type: "test"}, req)
-		So(err, ShouldBeNil)
+		require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/tsdb/request_test.go:41]:
```suggestion
		require.NoError(t, err)
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]

File: pkg/tsdb/request_test.go (hunk 1 above)
Comment (ceh) [reply pkg/tsdb/request_test.go]: 👍  Done.

File: pkg/tsdb/request_test.go (hunk 2 above)
Comment (ceh) [reply pkg/tsdb/request_test.go]: 👍  Done.

File: pkg/tsdb/request_test.go (hunk 3 above)
Comment (ceh) [reply pkg/tsdb/request_test.go]: 👍  Done.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Comment by aknuds1** [comment]
Thanks for cleaning things up!

**Pull request merged**